packages/ui/src/components/PreviewItem/PreviewItem.tsx

```diff
@@ -0,0 +1,66 @@
+import MagnifyingGlassPlusIcon from "@heroicons/react/24/outline/MagnifyingGlassPlusIcon";
+import { ReactNode } from "react";
+import { CardBasic } from "../Card";
+import Button from "../Button";
+
+export interface PreviewItemProps {
+  title: string;
+  details: ReactNode[];
+  buttonName: string;
+}
+
+const PreviewItem = ({ title, details, buttonName }: PreviewItemProps) => {
+  return (
+    <div
+      data-h2-position="base(relative)"
+      data-h2-display="base(grid)"
+      data-h2-grid-template-columns="base(3fr 1fr)"
+      data-h2-align-items="base(center)"
+      data-h2-background-color="base(foreground)"
+      data-h2-padding="base(x1)"
+    >
+      <div>
+        <p
+          data-h2-font-weight="base(bold)"
+          data-h2-text-decoration="base(underline)"
+          data-h2-margin-bottom="base(x.5)"
+        >
+          {title}
+        </p>
+        <div
+          data-h2-display="base(flex)"
+          data-h2-flex-wrap="base(wrap)"
+          data-h2-align-items="base(center)"
+          data-h2-gap="base(x.25, x.5)"
+        >
+          {details.map((detail, index) => {
+            const last = details.length;
+            return (
+              <>
+                <span
+                  data-h2-font-size="base(caption)"
+                  data-h2-color="base(grays)"
+                >
+                  {detail}
+                </span>
+                {index + 1 !== last && <span aria-hidden>&bull;</span>}
+              </>
+            );
+          })}
+        </div>
+      </div>
+      <Button
```

Comment:
I have bad vibes about the a11y of this but I feel like I don't know enough to be certain.  I might have to phone a friend. :sweat_smile:  @esizer could you please help us out?

- I have warning from Firefox saying "[Clickable elements must be focusable and should have interactive semantics](https://developer.mozilla.org/en-US/docs/Web/Accessibility/Understanding_WCAG/Keyboard?utm_source=devtools&utm_medium=a11y-panel-checks-keyboard#clickable_elements_must_be_focusable_and_should_have_interactive_semantics)".  The button is focusable but the warning is for the whole component, so I'm not sure.
- I feel like this item should have an aria role or something more specific that div.  Button?  li?  :shrug: 
- Is it a violation for the title to be underlined but not be a link?
- Should the title be a header element?
- Should the items be li's in a ul?

Reply:
> I have warning from Firefox saying "[Clickable elements must be focusable and should have interactive semantics](https://developer.mozilla.org/en-US/docs/Web/Accessibility/Understanding_WCAG/Keyboard?utm_source=devtools&utm_medium=a11y-panel-checks-keyboard#clickable_elements_must_be_focusable_and_should_have_interactive_semantics)". The button is focusable but the warning is for the whole component, so I'm not sure.

I am getting a similar error and I can't tell why :confused: I kind riffed on this last night with Yoni. It should be fine :woman_shrugging: I have no clue why firefox seems to think the wrapping `div` has a click event on it :face_with_spiral_eyes: 

> Is it a violation for the title to be underlined but not be a link?

It is odd but not necessarily a violation. Honestly, making these entire things clickable is [not necessarily good practice](https://adrianroselli.com/2020/02/block-links-cards-clickable-regions-etc.html#Demo:~:text=marketers%20often%20want%20a%20larger%20click%20area%20around%20their%20calls%20to%20action%20(CTAs)%20to%20make%20them%20more%20prominent.%20Without%20thinking%20about%20the%20impact%20on%20screen%20reader%20users) to be doing this but we can do it.

> Should the title be a header element?

Yeah, I'd feel more comfortable if it were a heading. Good call on that.

> I feel like this whole component should have an aria role or something more specific that div. Button? li? 🤷
> Should the items be li's in a ul?

From what I have seen, these are intended to be in lists all the time correct? If so then we probably should do something like what we have done for the description list.

```tsx

type PreivewItemElement = ReactElement<PreviewItem>;

type PreviewListProps extends DetailedHTMLProps<
  HtmlAttributes<HTMLULListElement>
  HTMLULListElement
> = {
  children: PreviewItemElement | Array<PreviewItemElement>; // Restircts children to only expected items
};

<PreivewList>
  <PReviewList.Item  {...props} />
</PreviewList>
```

Reply:
Okay I should have fixed most of this:
- Switched the PreviewItem to PreviewList using the compound component structure
- Changed the `<p/>` to <Heading />`
- This will be used as a list throughout the site, so the `<ul/>` was used for more semantic HTML
- The "[Clickable elements must be focusable and should have interactive semantics](https://developer.mozilla.org/en-US/docs/Web/Accessibility/Understanding_WCAG/Keyboard?utm_source=devtools&utm_medium=a11y-panel-checks-keyboard#clickable_elements_must_be_focusable_and_should_have_interactive_semantics)" is appearing throughout all the buttons on the site, so hopefully it's Firefox over evaluating. However, it's something we should confirm right away.